app/src/main/java/com/fpvout/digiview/MainActivity.java

```diff
@@ -163,94 +167,78 @@ public boolean onTouchEvent(MotionEvent event) {
         return super.onTouchEvent(event);
     }
 
-    private void updateWatermark() {
-        if (overlayView.getVisibility() == View.VISIBLE) {
-            watermarkView.setAlpha(0);
-            return;
-        }
+    private void toggleFullOverlay() {
+        if (overlayView.getAlpha() > 0.0f) return;
 
-        if (sharedPreferences.getBoolean(ShowWatermark, true)) {
-            watermarkView.setAlpha(0.3F);
-        } else {
-            watermarkView.setAlpha(0F);
-        }
+        toggleView(settingsButton, hideButtonsRunnable);
     }
 
-    private void updateVideoZoom() {
-        if (sharedPreferences.getBoolean(VideoZoomedIn, true)) {
-            mVideoReader.zoomIn();
-        } else {
-            mVideoReader.zoomOut();
-        }
-    }
+    private void hideFullOverlay() {
+        toggleView(watermarkView, sharedPreferences.getBoolean(ShowWatermark, true), 0.3f);
 
-    private void cancelButtonAnimation() {
-        Handler handler = settingsButton.getHandler();
-        if (handler != null) {
-            handler.removeCallbacksAndMessages(null);
-        }
+        toggleView(settingsButton, false);
+        toggleView(overlayView, false);
     }
 
-//    private void showSettingsButton() {
-//        cancelButtonAnimation();
-//
-//        if (overlayView.getVisibility() == View.VISIBLE) {
-//            buttonAlpha = 1;
-//            settingsButton.setAlpha(1);
-//        }
-//    }
+    private void showFullOverlay() {
```

Comment:
maybe moving this to OverlayView would make it  cleaner.
and the toggleViews function to a CustomView class, that Overlay could inherit from ?